### ISSUE TITLE
[SDPA] iree-compile fails to determine a tile size on SDPA with head_dim=1 for rocm/gfx942

### ISSUE THREAD

**Issue by rkayaith** [iree-org/iree#24319]
`iree-compile` fails to determine a tile size on `torch.aten.scaled_dot_product_attention` with `head_dim=1` for rocm/gfx942.

## Repro

`sdpa_head_dim_1.mlir`:

```mlir
module @module {
  func.func @main(%o_3_: !torch.tensor<[1,4,2,1],f32>, %k_1: !torch.vtensor<[1,3,2,1],f32>, %q_0: !torch.vtensor<[1,4,2,1],f32>, %v_2: !torch.vtensor<[1,3,2,1],f32>) attributes {torch.assume_strict_symbolic_shapes} {
    %permute_Q_val_0_sdpa_fprop = torch.constant.int 0
    %permute_Q_val_1_sdpa_fprop = torch.constant.int 2
    %permute_Q_val_2_sdpa_fprop = torch.constant.int 1
    %permute_Q_val_3_sdpa_fprop = torch.constant.int 3
    %permute_Q_sdpa_fprop = torch.prim.ListConstruct %permute_Q_val_0_sdpa_fprop, %permute_Q_val_1_sdpa_fprop, %permute_Q_val_2_sdpa_fprop, %permute_Q_val_3_sdpa_fprop : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %q_0_sdpa_fprop_perm = torch.aten.permute %q_0, %permute_Q_sdpa_fprop : !torch.vtensor<[1,4,2,1],f32>, !torch.list<int> -> !torch.vtensor<[1,2,4,1],f32>

    %permute_K_val_0_sdpa_fprop = torch.constant.int 0
    %permute_K_val_1_sdpa_fprop = torch.constant.int 2
    %permute_K_val_2_sdpa_fprop = torch.constant.int 1
    %permute_K_val_3_sdpa_fprop = torch.constant.int 3
    %permute_K_sdpa_fprop = torch.prim.ListConstruct %permute_K_val_0_sdpa_fprop, %permute_K_val_1_sdpa_fprop, %permute_K_val_2_sdpa_fprop, %permute_K_val_3_sdpa_fprop : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %k_1_sdpa_fprop_perm = torch.aten.permute %k_1, %permute_K_sdpa_fprop : !torch.vtensor<[1,3,2,1],f32>, !torch.list<int> -> !torch.vtensor<[1,2,3,1],f32>

    %permute_V_val_0_sdpa_fprop = torch.constant.int 0
    %permute_V_val_1_sdpa_fprop = torch.constant.int 2
    %permute_V_val_2_sdpa_fprop = torch.constant.int 1
    %permute_V_val_3_sdpa_fprop = torch.constant.int 3
    %permute_V_sdpa_fprop = torch.prim.ListConstruct %permute_V_val_0_sdpa_fprop, %permute_V_val_1_sdpa_fprop, %permute_V_val_2_sdpa_fprop, %permute_V_val_3_sdpa_fprop : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %v_2_sdpa_fprop_perm = torch.aten.permute %v_2, %permute_V_sdpa_fprop : !torch.vtensor<[1,3,2,1],f32>, !torch.list<int> -> !torch.vtensor<[1,2,3,1],f32>

    %none_mask_sdpa_fprop = torch.constant.none
    %dropout_sdpa_fprop = torch.constant.float 0.000000e+00
    %is_causal_sdpa_fprop = torch.constant.bool false
    %scale_sdpa_fprop = torch.constant.float 1.000000e+00
    %enable_gqa_sdpa_fprop = torch.constant.bool false
    %o_3_sdpa_fprop_perm = torch.aten.scaled_dot_product_attention %q_0_sdpa_fprop_perm, %k_1_sdpa_fprop_perm, %v_2_sdpa_fprop_perm, %none_mask_sdpa_fprop, %dropout_sdpa_fprop, %is_causal_sdpa_fprop, %scale_sdpa_fprop, %enable_gqa_sdpa_fprop : !torch.vtensor<[1,2,4,1],f32>, !torch.vtensor<[1,2,3,1],f32>, !torch.vtensor<[1,2,3,1],f32>, !torch.none, !torch.float, !torch.bool, !torch.float, !torch.bool -> !torch.vtensor<[1,2,4,1],f32>

    %permute_O_val_0_sdpa_fprop = torch.constant.int 0
    %permute_O_val_1_sdpa_fprop = torch.constant.int 2
    %permute_O_val_2_sdpa_fprop = torch.constant.int 1
    %permute_O_val_3_sdpa_fprop = torch.constant.int 3
    %permute_O_sdpa_fprop = torch.prim.ListConstruct %permute_O_val_0_sdpa_fprop, %permute_O_val_1_sdpa_fprop, %permute_O_val_2_sdpa_fprop, %permute_O_val_3_sdpa_fprop : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %o_3 = torch.aten.permute %o_3_sdpa_fprop_perm, %permute_O_sdpa_fprop : !torch.vtensor<[1,2,4,1],f32>, !torch.list<int> -> !torch.vtensor<[1,4,2,1],f32>

    torch.overwrite.tensor.contents %o_3 overwrites %o_3_ : !torch.vtensor<[1,4,2,1],f32>, !torch.tensor<[1,4,2,1],f32>
    return
  }
}
```

```bash
iree-compile sdpa_head_dim_1.mlir --iree-hal-target-backends=rocm --iree-rocm-target=mi300x
```

## Error output

```
sdpa_head_dim_1.mlir:35:28: error: 'linalg.generic' op tile size analysis did not determine a valid tile size
    %o_3_sdpa_fprop_perm = torch.aten.scaled_dot_product_attention %q_0_sdpa_fprop_perm, %k_1_sdpa_fprop_perm, %v_2_sdpa_fprop_perm, %none_mask_sdpa_fprop, %dropout_sdpa_fprop, %is_causal_sdpa_fprop, %scale_sdpa_fprop, %enable_gqa_sdpa_fprop : !torch.vtensor<[1,2,4,1],f32>, !torch.vtensor<[1,2,3,1],f32>, !torch.vtensor<[1,2,3,1],f32>, !torch.none, !torch.float, !torch.bool, !torch.float, !torch.bool -> !torch.vtensor<[1,2,4,1],f32>
                           ^
sdpa_head_dim_1.mlir:35:28: note: see current operation:
%30 = "linalg.generic"(%0, %20) <{indexing_maps = [affine_map<(d0, d1, d2) -> ()>, affine_map<(d0, d1, d2) -> (d1, d2, d0)>], iterator_types = [#linalg.iterator_type<parallel>, #linalg.iterator_type<parallel>, #linalg.iterator_type<parallel>], operandSegmentSizes = array<i32: 1, 1>}> ({
^bb0(%arg43: f32, %arg44: f32):
  %79 = "arith.mulf"(%arg43, %arg44) <{fastmath = #arith.fastmath<none>}> : (f32, f32) -> f32
  "linalg.yield"(%79) : (f32) -> ()
}) : (f32, tensor<1x1x1xf32>) -> tensor<1x1x1xf32>
sdpa_head_dim_1.mlir:35:28: error: failed to run translation of source executable to target executable for backend #hal.executable.target<"rocm", "rocm-hsaco-fb", ...>
```

## Environment

- Hardware: MI300X (gfx942)
- IREE: built from 0380544b17

**Comment by keshavvinayak01** [comment]
This is the same issue as https://github.com/iree-org/iree/pull/24239. That patch is now reverted because of an edge case. Once that is in this should be resolved.

**Comment by sommerlukas** [comment]
With the changes landed in https://github.com/iree-org/iree/pull/24243, https://github.com/iree-org/iree/pull/24431, https://github.com/iree-org/iree/pull/24449 and https://github.com/iree-org/iree/pull/24408, these cases should now be handled properly through masking. 

After landing https://github.com/iree-org/iree/pull/24408, the reproducers from this issue compile and compute correctly for me with `main`.
